app/src/main/java/edu/drexel/lapcounter/lapcounter/backend/Database/Device/DeviceDao.java

```diff
@@ -0,0 +1,30 @@
+package edu.drexel.lapcounter.lapcounter.backend.Database.Device;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.persistence.room.Dao;
+import android.arch.persistence.room.Insert;
+import android.arch.persistence.room.Query;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import edu.drexel.lapcounter.lapcounter.backend.Database.Workout.Workouts;
+
+
+@Dao
+public interface DeviceDao {
+
+    @Insert
+    void addDevice(Device device);
+
+    @Query("SELECT * FROM devices")
+    List<Device> getAllDevices();
```

Comment:
If we made this guy return `LiveData<List<Device>>`, then we could probably avoid the all `ExecutorService` and related threading business in the DeviceRepository.

Reply:
It would, but I found that for the use cases that we have already we are never worried about the data within the database updating, just what is in it at the given moment.  Also, LiveData does not seem to have an easy way to access the data that's in it at any one moment because its expecting changes, so I found a List worked best.